### PR TITLE
compute the matrix of an isogeny on a given n-torsion subgroup

### LINKED ISSUE #34879: compute matrix of isogeny on n-torsion
<div id="comment:0"></div>

The task of computing the matrix by which an isogeny acts on a particular n‑torsion subgroup often comes up in algorithms for supersingular elliptic curves. (In particular, it appears as a subroutine of computing the kernel of an endomorphism-ring ideal.)

This patch adds a simple method to compute such a matrix, based on `AdditiveAbelianGroupWrapper.discrete_log()`.

(The dependency is only necessary for the reference in the "see also" block.)

Depends on #34718

Component: **elliptic curves**

Author: **Lorenz Panny**

Branch/Commit: **[public/matrix_of_isogeny](https://github.com/sagemath/sagetrac-mirror/tree/public/matrix_of_isogeny) @ [`0bc249d`](https://github.com/sagemath/sagetrac-mirror/commit/0bc249d5c71c5c74356fc3752cfb3caf9169145e)**

_Issue created by migration from https://trac.sagemath.org/ticket/34879_

### PR DESCRIPTION
**Pull request by yyyyx4**
Fixes #34879.

### CODE REVIEW

**Comment by JohnCremona** [comment]
OK, I will review this

File: src/sage/schemes/elliptic_curves/hom.py [hunk 1]
```diff
@@ -722,6 +724,108 @@ def as_morphism(self):
         Y_affine = Curve(self.codomain()).affine_patch(2)
         return X_affine.hom(self.rational_maps(), Y_affine).homogenize(2)
 
+    def matrix_on_subgroup(self, domain_gens, codomain_gens=None):
+        r"""
+        Return the matrix as which this isogeny acts on the
```
Comment (JohnCremona) [inline src/sage/schemes/elliptic_curves/hom.py]: "by which", not "as which"

File: src/sage/schemes/elliptic_curves/hom.py [hunk 2]
```diff
@@ -722,6 +724,108 @@ def as_morphism(self):
...
+        `n`-torsion subgroup with respect to the given bases.
+
+        (This is useful for tasks such as computing a set of
```
Comment (JohnCremona) [inline src/sage/schemes/elliptic_curves/hom.py]: I would leave out the parenthetical remark.  Other people might have other uses!

**Review by JohnCremona — CHANGES_REQUESTED** [review]

File: src/sage/schemes/elliptic_curves/hom.py (hunk 2 above)
Comment (JohnCremona) [reply src/sage/schemes/elliptic_curves/hom.py]: Basically this is a postive review, I just recommend one change of wording and make another minor suggestion, both in docstrings.

Comment (yyyyx4) [reply src/sage/schemes/elliptic_curves/hom.py]: Thanks for your comments! I converted the parenthetical remark to an example (making use of functionality recently merged in #34968), which is more helpful anyway.

Comment (JohnCremona) [reply src/sage/schemes/elliptic_curves/hom.py]: Perfect!

**Pull request merged**